File: Android.RestaurantMenu/app/src/main/java/com/bojinzhang/android/Business/CategoriesBuss.java
package com.bojinzhang.android.Business;

import com.bojinzhang.android.Model.CategoryModel;
import com.bojinzhang.android.Util.HttpUtil;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.ArrayList;

import okhttp3.Response;

/**
 * Created by zhangbojin on 29/05/17.
 */

public class CategoriesBuss {
    public ArrayList<CategoryModel> getCategories() {
        ArrayList<CategoryModel> categoryModels = new ArrayList<CategoryModel>();

        try {
            Response response = HttpUtil.sendHttpRequest("http://10.0.2.2:8081/api/Category/GetAllCategories");
            String data = response.body().string();

            JSONArray categories = new JSONObject(data).getJSONArray("Categories");
            for (int i = 0; i < categories.length(); i++) {
                JSONObject item = categories.getJSONObject(i);
                int id = item.getInt("Id");
                String eName = item.getString("EName");
                String oName = item.getString("OtherName");

                CategoryModel model = new CategoryModel(eName, oName, id);
                categoryModels.add(model);
            }
        } catch (Exception e) {

        }
        return categoryModels;
    }
}

File: Web/RestaurantOnline/Business.RestaurantOnline/CategoriesBuss.cs
using System;
using System.Data;
using System.Collections.Generic;
using Data.RestaurantOnline;
using Model.RestaurantOnline;

namespace Business.RestaurantOnline
{
    public class CategoriesBuss
    {
        CategoryDao dao = new CategoryDao();

        public List<CategoryModel> GetCategories()
        {
            var ds = dao.GetCategories();
            List<CategoryModel> categories = null;

            if (ds != null && ds.Tables.Count > 0 && ds.Tables[0].Rows.Count > 0)
            {
                categories = new List<CategoryModel>();
                foreach (DataRow dr in ds.Tables[0].Rows)
                {
                    var cm = new CategoryModel();
                    cm.Id = Int32.Parse(dr[0].ToString());
                    cm.EName = dr[1].ToString();
                    cm.OtherName = dr[2].ToString();

                    categories.Add(cm);
                }
            }

            return categories;
        }
    }
}
